src/Clapi/Valuespace.hs

```diff
@@ -305,31 +306,39 @@ rectifyTypes vs explicitPaths =
     populateDefsFor deflessNodes vs = if null deflessNodes then (mempty, vs) else let
         np = head $ Set.toList deflessNodes
       in do
-        vs' <- case getOrBuildDef deflessNodes np of
-            Left err -> (Map.fromList [(np, err)], vs)
-            Right (tp, md) ->
-                ( mempty
-                , over types (Mos.setDependency np tp) $ over defs (Map.insert tp md) vs)
-        populateDefsFor (Set.delete np deflessNodes) vs'
-
-    getOrInferType :: Set.Set NodePath -> NodePath -> CanFail TypePath
-    getOrInferType dirtyPaths np = if Set.member np dirtyPaths
-      then maybe (parentDerivedType dirtyPaths np) Right $ Map.lookup np explicitlyAssigned
-      else getType vs np
-    parentDerivedType :: Set.Set NodePath -> NodePath -> CanFail TypePath
-    parentDerivedType dirtyPaths np = case splitBasename np of
-        Just (pp, cn) ->
-            getOrBuildDef dirtyPaths pp >>=
-            return . snd >>=
-            note "Parent has no type for child" . flip childTypeFor cn
-        Nothing -> error "Hit root deriving parent types, code bad."
-    getOrBuildDef :: Set.Set NodePath -> NodePath -> CanFail (TypePath, Definition)
-    getOrBuildDef dirtyPaths np = do
-        tp <- getOrInferType dirtyPaths np
-        mt <- getOrInferType dirtyPaths tp >>= categoriseMetaTypePath
-        tn <- getNode tp vs
-        d <- validateTypeNode mt tn
-        return (tp, d)
+        case runState (runExceptT (getOrBuildDef np)) (deflessNodes, vs) of
+            (Left err, (deflessNodes', vs')) -> (Map.fromList [(np, err)], vs)
```

Comment:
You don't used `(deflessNodes', vs')` here, so you could either replace them or prefix them with `_`. It'll keep the compiler happy if you switch `-Wall` on, and makes it clearer you're ignoring stuff ;-)